app/services/articles/builder.rb

```diff
@@ -6,87 +6,79 @@ def initialize(user, tag, prefill)
       @user = user
       @tag = tag
       @prefill = prefill
-    end
-
-    def build
-      if @tag.present? && @user&.editor_version == "v2"
-
-        [tag_user_editor_v2, needs_authorization]
-      elsif @tag&.submission_template.present? && @user
-
-        [tag_user, needs_authorization]
-      elsif @prefill.present? && @user&.editor_version == "v2"
-
-        [prefill_user_editor_v2, needs_authorization]
-      elsif @prefill.present? && @user
 
-        [prefill_user, needs_authorization]
-      elsif @tag.present?
-
-        [tag, does_not_need_authorization]
-      else
-        return [user_editor_v2, does_not_need_authorization] if @user&.editor_version == "v2"
-
-        [user_editor_v1, does_not_need_authorization]
-      end
+      @editor_version2 = @user&.editor_version == "v2"
     end
 
-    def needs_authorization
-      true
+    def self.call(*args)
+      new(*args).call
     end
 
-    def does_not_need_authorization
-      false
+    # the Builder returns a pair of [article, needs_authorization?]
+    # => `needs_authorization? can be either true or false
+    def call
```

Comment:
Whew this is a trip!

Reply:
yeah, I still don't fully understand it :D